### PR TITLE
Add httptrace integration.

### PR DESCRIPTION
**Pull request by anmic**

### CODE REVIEW

**Comment by vmihailenco** [comment]
You should create `ClientTrace` and inject it into context using https://golang.org/pkg/net/http/httptrace/#WithClientTrace in https://github.com/airbrake/gobrake/blob/master/route_trace.go#L18-L28

**Comment by anmic** [comment]
Inject `ClientTrace` into context.

File: route_trace.go [hunk 1]
```diff
@@ -15,6 +16,17 @@ type RouteTrace struct {
...
 var _ Trace = (*RouteTrace)(nil)
 
+func newClientTrace(routeTrace *RouteTrace) *httptrace.ClientTrace {
+	return &httptrace.ClientTrace{
+		ConnectStart: func(network, addr string) {
+			routeTrace.StartSpan("request")
+		},
+		ConnectDone: func(network, addr string, err error) {
+			routeTrace.EndSpan("request")
+		},
```
Comment (vmihailenco) [inline route_trace.go]: What does this measures?

**Comment by vmihailenco** [comment]
Add tests

**Comment by anmic** [comment]
Add test.

**Pull request closed without merging**